lib/android_apk.rb

```diff
@@ -421,28 +455,11 @@ def self.read_signature(apk, filepath)
     end
   end
 
-  def self.read_adaptive_icon(apk, filepath)
-    # candidate
-    adaptive_icon_path = apk.icon or return
-
-    return unless adaptive_icon_path.end_with?(".xml")
+  # @return [AndroidApk::Xmltree, NilClass]
+  private def icon_xmltree
+    return @icon_xmltree if defined?(@icon_xmltree)
 
-    png_dpi = SUPPORTED_DPI_NAMES.find { |d| apk.icon_path_hash[d]&.end_with?(".png") }
-
-    # at least one png icon is required if min sdk version doesn't support adaptive icon
-    if apk.min_sdk_version.to_i < ADAPTIVE_ICON_SDK && png_dpi.nil?
-      return
-    end
-
-    # invalid xml file may throw an error
-    Zip::File.open(filepath) do |zip_file|
-      apk.adaptive_icon = zip_file.find_entry(adaptive_icon_path)&.get_input_stream do |entry|
-        !!entry.read.include?("adaptive-icon")
```

Comment:
this logic worked even if the xml is decoded but the result included false-positives.